test/test_collection.py

```diff
@@ -2125,7 +2125,7 @@ def test_find_command_generation(self):
             None,
         )
         self.assertEqual(
-            cmd.to_dict(), SON([("find", "coll"), ("$dumb", 2), ("filter", {"foo": 1})]).to_dict()
+            cmd, SON([("find", "coll"), ("$dumb", 2), ("filter", {"foo": 1})]).to_dict()
```

Comment:
Hmm, this highlights how this is a subtle breaking change since the command payloads published in CommandStartedEvent will change from SON to dict. Code relying on SON will break when it changes to dict.

We should at least document in the changelog and mark the jira ticket as a minor breaking change.

Reply:
Also since this is the case I revise my earlier opinion. I think it's fine to change PoolOptions.metadata type.

Reply:
If we're removing SON entirely, this test case should compare against an actual `dict`, not a converted `SON`.